lib/kubernetes-deploy/kubernetes_resource/pod.rb

```diff
@@ -89,6 +70,27 @@ def fetch_logs
 
     private
 
+    def interpret_pod_status_data(status_data, metadata)
+      @status = @phase = (metadata["deletionTimestamp"] ? "Terminating" : status_data["phase"])
+
+      if @phase == "Failed" && status_data['reason'].present?
+        @status += " (Reason: #{status_data['reason']})"
+      elsif @phase != "Terminating"
+        ready_condition = status_data.fetch("conditions", []).find { |condition| condition["type"] == "Ready" }
+        @ready = ready_condition.present? && (ready_condition["status"] == "True")
+        @status += " (Ready: #{@ready})"
```

Comment:
Yeah, my third branch in this stack (the one you haven't seen yet) should cover this. 👍